src/gui/attributetable/qgsdualview.cpp

```diff
@@ -85,6 +85,8 @@ void QgsDualView::init( QgsVectorLayer* layer, QgsMapCanvas* mapCanvas, const Qg
   else
     columnBoxInit();
 
+  mTableView->resizeColumnsToContents();
```

Comment:
Reimplementing QTableView::sizeHintForColumn is probably the way to go...

On 08/31/2015 02:09 PM, Alessandro Pasotti wrote:
>
> In src/gui/attributetable/qgsdualview.cpp
> <https://github.com/qgis/QGIS/pull/2259#discussion_r38305538>:
>
> > @@ -85,6 +85,8 @@ void QgsDualView::init( QgsVectorLayer* layer, QgsMapCanvas* mapCanvas, const Qg
> >    else
> >      columnBoxInit();
> >  
> > +  mTableView->resizeColumnsToContents();
>
> @m-kuhn <https://github.com/m-kuhn> yes, the best solution would
> probably be adjusting on visible rows, unfortunately I don't know when
> I'll have some more time to investigate on this. For the time being
> it's better to delete that line (maybe leaving a TODO)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2259/files#r38305538>.
>